### ISSUE TITLE
SQLite: Cannot parse "insert or" statements (ABORT/FAIL/IGNORE/REPLACE/ROLLBACK)

### ISSUE THREAD

**Issue by imrehg** [sqlfluff/sqlfluff#3733]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The SQLite dialect doesn't seem to recognize variants of the `INSERT` statement where where insert failure behaviours are defined with an `OR ...` statement. The options are [available in the docs](https://www.sqlite.org/lang_insert.html).

```sql
INSERT OR ABORT INTO ...
INSERT OR FAIL INTO ...
INSERT OR IGNORE INTO ...
INSERT OR REPLACE INTO ...
INSERT OR ROLLBACK INTO ...
```

### Expected Behaviour

```shell
$ echo 'CREATE TABLE T1 (ID integer PRIMARY KEY); INSERT OR ABORT INTO T1 (ID) VALUES (1); INSERT OR FAIL INTO T1 (ID) VALUES (2); INSERT OR IGNORE INTO T1 (ID) VALUES (3); INSERT OR REPLACE INTO T1 (ID) VALUES (4); INSERT OR ROLLBACK INTO T1 (ID) VALUES (5);' | sqlfluff lint --dialect sqlite -
All Finished 📜 🎉!
```

### Observed Behaviour

```shell
$ echo 'CREATE TABLE T1 (ID integer PRIMARY KEY); INSERT OR ABORT INTO T1 (ID) VALUES (1); INSERT OR FAIL INTO T1 (ID) VALUES (2); INSERT OR IGNORE INTO T1 (ID) VALUES (3); INSERT OR REPLACE INTO T1 (ID) VALUES (4); INSERT OR ROLLBACK INTO T1 (ID) VALUES (5);' | sqlfluff lint --dialect sqlite -
== [stdin] FAIL                                                                                                
L:   1 | P:  43 |  PRS | Line 1, Position 43: Found unparsable section: 'INSERT
                       | OR ABORT INTO T1 (ID) VALUES (1)'
L:   1 | P:  84 |  PRS | Line 1, Position 84: Found unparsable section: 'INSERT
                       | OR FAIL INTO T1 (ID) VALUES (2)'
L:   1 | P: 124 |  PRS | Line 1, Position 124: Found unparsable section: 'INSERT
                       | OR IGNORE INTO T1 (ID) VALUES (3)...'
L:   1 | P: 166 |  PRS | Line 1, Position 166: Found unparsable section: 'INSERT
                       | OR REPLACE INTO T1 (ID) VALUES (4...'
L:   1 | P: 209 |  PRS | Line 1, Position 209: Found unparsable section: 'INSERT
                       | OR ROLLBACK INTO T1 (ID) VALUES (...'
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```shell
echo 'CREATE TABLE T1 (ID integer PRIMARY KEY); INSERT OR ABORT INTO T1 (ID) VALUES (1); INSERT OR FAIL INTO T1 (ID) VALUES (2); INSERT OR IGNORE INTO T1 (ID) VALUES (3); INSERT OR REPLACE INTO T1 (ID) VALUES (4); INSERT OR ROLLBACK INTO T1 (ID) VALUES (5);' | sqlfluff lint --dialect sqlite -
```

### Dialect

`sqlite`

### Version

```shell
$ sqlfluff --version
sqlfluff, version 1.2.1
$ python -V
Python 3.10.5
```

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
